src/components/Button/Button.js

```diff
@@ -23,48 +23,47 @@ const propTypes = {
   onClick: PropTypes.func.isRequired,
 };
 
+const NOOP = () => {};
+
 const Button = props => {
   const [isElementDisabled, customOverrides] = useSelector(state => [
     selectors.isElementDisabled(state, props.dataElement),
     selectors.getCustomElementOverrides(state, props.dataElement),
   ]);
 
-  props = { ...props, ...customOverrides };
   const {
+    disable,
     isActive,
     mediaQueryClassName,
     img,
     label,
     color,
     dataElement,
-    onClick = () => {},
+    onClick = NOOP,
```

Comment:
why not let it have the value of undefined?

Reply:
Leave it as undefined can also work. If we don't have a default value for this then the code will be something like this I think:
```javascript
const { onClick } = { ...props, ...customOverrides };

onClick={(e) => {
  if (!disabled && onClick) {
    onClick(e);
  }
}}
```
If we have a default value then
```javascript
const { onClick = NOOP } = { ...props, ...customOverrides };

onClick={disable ? NOOP : onClick}
```

I think it's more of a personal perference to think which one is better(or cleaner). I can't I like which one more so if the current way works I'm just usually too lazy to change it :)